File: PSET2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    string str = get_string("Text: ");
    int n_letters = 0, n_words = 1, n_sentences = 0;
    for (int i = 0; i < strlen(str); i++)
    {
        if (str[i] == ' ')
        {
            n_words++;
        }
        else if (str[i] == '.' || str[i] == '?' || str[i] == '!')
        {
            n_sentences++;
        }
        if (isalpha(str[i]))
        {
            n_letters++;
        }
    }
    double L = (100.0 * n_letters) / n_words;
    double S = (100.0 * n_sentences) / n_words;
    int readablity = (0.0588 * L - 0.296 * S - 15.8) + 0.5;
    if (readablity >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (readablity < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", readablity);
    }
}

File: PSET6/Readability/readability.py
txt = input("Text: ")
n_letters = 0
n_words = 1
n_sentences = 0
for i in txt:
    if (i == ' '):
        n_words += 1
    elif (i == '.' or i == '?' or i == '!'):
        n_sentences += 1
    if (i.isalpha()):
        n_letters += 1

L = 100 * n_letters / n_words
S = 100 * n_sentences / n_words
readablity = round(0.0588 * L - 0.296 * S - 15.8)

if readablity >= 16:
    print("Grade 16+")
elif readablity < 1:
    print("Before Grade 1")
else:
    print(f"Grade {readablity}")
